code/modules/mob/dead/observer/orbit.dm

```diff
@@ -18,94 +27,145 @@
 
 /datum/orbit_menu/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
 	. = ..()
+
 	if(.)
 		return
+
 	switch(action)
-		if ("orbit")
+		if("orbit")
 			var/ref = params["ref"]
-			var/atom/movable/poi = (locate(ref) in GLOB.mob_list) || (locate(ref) in GLOB.poi_list)
-			if (poi == null)
-				. = TRUE
-				return
-			owner.ManualFollow(poi)
-			owner.reset_perspective(null)
+			var/auto_observe = params["auto_observe"]
+			var/atom/poi = SSpoints_of_interest.get_poi_atom_by_ref(ref)
+
+			if((ismob(poi) && !SSpoints_of_interest.is_valid_poi(poi, CALLBACK(src, .proc/validate_mob_poi))) \
+				|| !SSpoints_of_interest.is_valid_poi(poi))
+				to_chat(usr, span_notice("That point of interest is no longer valid."))
+				return TRUE
+
+			var/mob/dead/observer/user = usr
+			user.ManualFollow(poi)
+			user.reset_perspective(null)
 			if (auto_observe)
-				owner.do_observe(poi)
-			. = TRUE
-		if ("refresh")
-			update_static_data(owner, ui)
-			. = TRUE
-		if ("toggle_observe")
-			auto_observe = !auto_observe
-			if (auto_observe && owner.orbit_target)
-				owner.do_observe(owner.orbit_target)
-			else
-				owner.reset_perspective(null)
+				user.do_observe(poi)
+			return TRUE
+		if("refresh")
+			update_static_data(usr, ui)
```

Comment:
Is there any point to the refresh button anymore if it's all in ui_data now?

Reply:
Good point. It can safely be removed now.

Reply:
Removed.